### PR TITLE
Make socks5 optional, off by default

### PR DESCRIPTION
**Pull request by howardjohn**
This is a super security sensitive codepath, and mostly and experiment.
Make it opt in so we don't stumble into a CVE when we ship beta.

### CODE REVIEW

File: src/config.rs
Comment (bleggett) [inline src/config.rs:53]:
```suggestion
const ENABLE_SOCKS5: &str = "ENABLE_SOCKS5";
```

would prefer to not use the UNSTABLE_ prefix, env vars do not have stability guarantees anyway, nor does anyone setting them by hand on their own ztunnel instance.

File: src/proxy.rs
Comment (bleggett) [inline src/proxy.rs:171]: nit: `.and_then()` is a tad more concise here I think.

**Review by stevenctl — APPROVED** [review]
LGTM, hold if you want to address Ben's comments before merging.

**Pull request merged**
